src/overlays/actors/ovl_En_Encount2/z_en_encount2.c

```diff
@@ -21,26 +28,214 @@ const ActorInit En_Encount2_InitVars = {
+        this->switchFlag = -1;
+    }
+
+    if ((this->switchFlag >= 0) && (Actor_GetSwitchFlag(globalCtx, this->switchFlag) != 0)) {
+        Actor_MarkForDeath(this);
+        return;
+    }
+
+    this->collider.elements->dim.modelSphere.radius = 0x39;
+    this->collider.elements->dim.scale = 1.0f;
+    this->collider.elements->dim.modelSphere.center.x = 0;
+    this->collider.elements->dim.modelSphere.center.y = -4;
+    this->collider.elements->dim.modelSphere.center.z = 0;
+
+    this->dynaActor.actor.colChkInfo.damageTable = &damageTable;
+    func_808E16FC(this);
+}
+
+void EnEncount2_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnEncount2* this = THIS;
+    BgCheck_RemoveActorMesh(globalCtx, &globalCtx->colCtx.dyna, this->dynaActor.bgId);
+    Collider_DestroyJntSph(globalCtx, &this->collider);
+}
+
+// init extension func
+void func_808E16FC(EnEncount2* this) {
+    this->statePopped = 0;
+    this->actionFunc = &func_808E1714;
+}
+
+//actionfunc: idle floating
+void func_808E1714(EnEncount2* this, GlobalContext* globalCtx) {
+    this->oscillationAngle += 1500.0f;
+    this->dynaActor.actor.velocity.y = Math_SinS(this->oscillationAngle);
+    Math_ApproachF(&this->scale, 0.1f, 0.3f, 0.01f);
+    if (((this->collider.base.acFlags & AC_HIT) != 0) && (this->dynaActor.actor.colChkInfo.damageEffect == 0xE)) {
+        this->dynaActor.actor.colChkInfo.health = 0;
+        this->statePopped = 1;
+        this->actionFunc = func_808E17C4; // pop
+    }
+}
+
+//actionfunc: pop
+void func_808E17C4(EnEncount2* this, GlobalContext* globalCtx) {
+    s32 i;
+    Vec3f curPos;
+
+    Math_Vec3f_Copy(&curPos, &this->dynaActor.actor.world.pos);
+    curPos.y += 60.0f;
+    Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_CLEAR_TAG, curPos.x, curPos.y, curPos.z,
+       0xFF, 0xFF, 0xC8, 0x0001);
+
+    for(i = 0; i != 100; i++){
+        func_808E1A24(this, &curPos, 10); // init particle
+    }
+
+    Audio_PlayActorSound2(this, 0x2949);
+    this->deathTimer = 30;
+    this->actionFunc = func_808E18A8; // post-pop
+}
+
+//actionfunc: after pop, waiting for timer then death
+void func_808E18A8(EnEncount2* this, GlobalContext* globalCtx) {
+    if (this->deathTimer == 0) {
+        if (this->switchFlag >= 0) {
+            Actor_SetSwitchFlag(globalCtx, this->switchFlag);
+        }
+        Actor_MarkForDeath(&this->dynaActor.actor);
+    }
+}
+
+void EnEncount2_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnEncount2* this = THIS;
+    s8 pad[4];
+
+    if (this->deathTimer != 0) {
+        this->deathTimer--;
+    }
+
+    this->dynaActor.actor.shape.rot.y = this->dynaActor.actor.world.rot.y;
+    Actor_SetHeight(&this->dynaActor.actor, 30.0f);
+    Actor_SetScale(&this->dynaActor.actor, this->scale);
+    this->actionFunc(this, globalCtx);
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->dynaActor.actor);
+    func_808E1B4C(this, globalCtx); // update all particles
+
+    if (! this->statePopped) {
```

Comment:
Boolean variables used like this are generally more easily read with a name like `isStatePopped`